cmd/dockerd/daemon.go

```diff
@@ -573,9 +574,13 @@ func loadListeners(cli *DaemonCli, serverConfig *apiserver.Config) ([]string, er
 		proto := protoAddrParts[0]
 		addr := protoAddrParts[1]
 
-		// It's a bad idea to bind to TCP without tlsverify.
+		// It's a *very* bad idea to bind to TCP without tlsverify.
 		if proto == "tcp" && (serverConfig.TLSConfig == nil || serverConfig.TLSConfig.ClientAuth != tls.RequireAndVerifyClientCert) {
-			logrus.Warn("[!] DON'T BIND ON ANY IP ADDRESS WITHOUT setting --tlsverify IF YOU DON'T KNOW WHAT YOU'RE DOING [!]")
+			if !cli.Config.InsecureHostBind {
+				return nil, fmt.Errorf("refusing to bind to an IP address without --tlsverify configured: use --give-the-internet-root-access to ignore this error")
```

Comment:
This will break existing deployments.
Probably we should have some grace period? (cc @thaJeztah )

Or maybe we could inject `time.Sleep(punishment)`

Reply:
> This will break existing deployments.

Well, yes. But I'm not sure how to gracefully deny it -- adding a sleep would not be noticed (people would probably blame systemd) and nobody has noticed the warning message on startup. We could make `docker` (the client) output a warning on each operation but that will also break deployments. (Personally I think it should've been done this way in the first place.)

Reply:
This would definitely require a deprecation period; one thing to make this situation more visible would be to include a warning to `docker info`, if the CLI is using an insecure TCP connection.

Also; making it easier to do the right thing (which could be, e.g. automatically generating a certificate) would help moving things to a better place